pkg/extensions/tbtc/recovery/recovery_test.go

```diff
@@ -102,6 +114,189 @@ func TestBuildSignedTransactionHexString(t *testing.T) {
+	keep := tbtcHandle.OpenKeep(keepAddress, depositAddress, memberAddresses)
+
+	btcAddresses := []string{
+		"1MjCqoLqMZ6Ru64TTtP16XnpSdiE8Kpgcx",
+		"3EktnHQD7RiAE6uzMj2ZifT9YgRrkSgzQX",
+		"bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq",
+	}
+	maxFeePerVByte := int32(73)
+
+	mutex := &sync.RWMutex{}
+
+	btcTransactions := make(map[string]string)
+
+	var providersInitializedWg sync.WaitGroup
+	providersInitializedWg.Add(groupSize)
+
+	for i, memberID := range groupMembers {
+		go func(memberID tss.MemberID, index int) {
+			memberPublicKey, err := memberID.PublicKey()
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			memberNetworkKey := key.NetworkPublic(*memberPublicKey)
+			networkProvider := local.ConnectWithKey(&memberNetworkKey)
+
+			providersInitializedWg.Done()
+			providersInitializedWg.Wait()
+
+			defer waitGroup.Done()
+
+			preParams := testData[index].LocalPreParams
+
+			signer, err := tss.GenerateThresholdSigner(
+				ctx,
+				keep.ID().Hex(),
+				memberID,
+				groupMembers,
+				uint(len(groupMembers)-1),
+				networkProvider,
+				pubKeyToAddressFn,
+				params.NewBox(&preParams),
+			)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			signedBtcTransaction, err := BuildBitcoinTransaction(
+				ctx,
+				networkProvider,
+				localChain,
+				tbtcHandle,
+				keep,
+				signer,
+				&chaincfg.MainNetParams,
+				btcAddresses,
+				maxFeePerVByte,
+			)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			mutex.Lock()
+			btcTransactions[memberID.String()] = signedBtcTransaction
+			mutex.Unlock()
+		}(memberID, i)
+	}
+
+	go func() {
+		waitGroup.Wait()
+		cancel()
+	}()
+
+	select {
+	case <-ctx.Done():
+		if len(btcTransactions) != groupSize {
+			t.Errorf(
+				"invalid number of results\nexpected: [%d]\nactual:  [%d]",
+				groupSize,
+				len(btcTransactions),
+			)
+		}
+
+		expectedSignature := btcTransactions[groupMembers[0].String()]
+		expectedOutputValue := int64(3329050) // (original deposit of 10000000 - fee) / 3
+		originalDepositSats := int64(10000000)
+
+		for _, memberID := range groupMembers {
+			if memberResult, ok := btcTransactions[memberID.String()]; ok {
+				if memberResult != expectedSignature {
+					t.Errorf(
+						"hex strings must all be identical\nexpected: %s\nactual:   %s",
+						expectedSignature,
+						memberResult,
+					)
+				}
+
+				decodedTransaction := decodeTransaction(t, memberResult)
+
+				if len(decodedTransaction.TxOut) != 3 {
+					t.Errorf("wrong number of outputs\nexpected: 1\nactual:   %d", len(decodedTransaction.TxOut))
+				}
+
+				for _, outputTransaction := range decodedTransaction.TxOut {
+					if outputTransaction.Value != expectedOutputValue {
+						t.Errorf(
+							"incorrect output value\nexpected: %d\nactual:   %d",
+							expectedOutputValue,
+							outputTransaction.Value,
+						)
+					}
+				}
+
+				validateTransaction(t, decodedTransaction, originalDepositSats) // original deposit amount
```

Comment:
We could do this validation outside of the loop for the first result only, as we've already confirmed all the results are the same hex strings.